plugins/askrene/layer.c

```diff
@@ -223,48 +216,41 @@ const struct local_channel *layer_find_local_channel(const struct layer *layer,
 }
 
 static struct constraint *layer_find_constraint_nonconst(const struct layer *layer,
-							 const struct short_channel_id_dir *scidd,
-							 enum constraint_type type)
+							 const struct short_channel_id_dir *scidd)
 {
-	struct constraint_key k = { *scidd, type };
-	return constraint_hash_get(layer->constraints, &k);
+	return constraint_hash_get(layer->constraints, scidd);
 }
 
 /* Public one returns const */
 const struct constraint *layer_find_constraint(const struct layer *layer,
-					       const struct short_channel_id_dir *scidd,
-					       enum constraint_type type)
+					       const struct short_channel_id_dir *scidd)
 {
-	return layer_find_constraint_nonconst(layer, scidd, type);
+	return layer_find_constraint_nonconst(layer, scidd);
 }
 
 const struct constraint *layer_update_constraint(struct layer *layer,
 						 const struct short_channel_id_dir *scidd,
-						 enum constraint_type type,
 						 u64 timestamp,
-						 struct amount_msat limit)
+						 const struct amount_msat *min,
+						 const struct amount_msat *max)
 {
-	struct constraint *c = layer_find_constraint_nonconst(layer, scidd, type);
+	struct constraint *c = layer_find_constraint_nonconst(layer, scidd);
```

Comment:
Why not create a new constraint every time?
I can see that this would seem unnecessary if we have to put them together anyways when we compute the routes.

But the problem is that we combine an old constraint with a new one and use the time stamp of the youngest.
If the old constraint is putting extra information on the new constraint we might be saying that whatever the age
of that old information, now it is brand new, ie. we don't give a change to old constraints to become old we just refresh their time stamp every time we add new information on the same channel.

For example, last week I learned that this channel has max=1000sat, today I learned that max=11000sat, the
new combined constrained is still the value of the old constraint and it will have the time stamp of today. But those
bits of information where learned already a week ago and we are dragging them like if we had learned about it
today. They will never expire.

Reply:
Also one might argue that anyways that if you use `getroutes` you will never probe a channel with an amount
greater than the max of constraints, so there is no way we might learn max=1100sat today.
But why not probe other amounts besides what `getroutes` suggests?

Reply:
Agreed, this was premature optimization!  I'll just accumulate them for now, and we can get smarter later if they get too unwieldy.